File: alphatimer-protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/messages/DataHandlingMessage1.java
package org.lisasp.alphatimer.api.protocol.events.messages;

import lombok.Value;
import org.lisasp.alphatimer.api.protocol.events.messages.enums.KindOfTime;
import org.lisasp.alphatimer.api.protocol.events.messages.enums.MessageType;
import org.lisasp.alphatimer.api.protocol.events.messages.enums.RankInfo;
import org.lisasp.alphatimer.api.protocol.events.messages.enums.TimeType;
import org.lisasp.alphatimer.api.protocol.events.messages.values.UsedLanes;

import java.time.LocalDateTime;

@Value
public class DataHandlingMessage1 implements Message {
    private final LocalDateTime timestamp;
    private final String competition;
    private final String original;
    private final MessageType messageType;
    private final KindOfTime kindOfTime;
    private final TimeType timeType;
    private final UsedLanes usedLanes;
    private final byte lapCount;
    private final short event;
    private final byte heat;
    private final byte rank;
    private final RankInfo rankInfo;

}

File: alphatimer-protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/messages/DataHandlingMessage1.kt
package org.lisasp.alphatimer.api.protocol.events.messages

import org.lisasp.alphatimer.api.protocol.events.messages.enums.KindOfTime
import org.lisasp.alphatimer.api.protocol.events.messages.enums.MessageType
import org.lisasp.alphatimer.api.protocol.events.messages.enums.RankInfo
import org.lisasp.alphatimer.api.protocol.events.messages.enums.TimeType
import org.lisasp.alphatimer.api.protocol.events.messages.values.UsedLanes
import java.time.LocalDateTime

data class DataHandlingMessage1(
    val original: String,
    val messageType: MessageType,
    val kindOfTime: KindOfTime,
    val timeType: TimeType,
    val usedLanes: UsedLanes,
    val lapCount: Byte,
    val event: Short, val heat: Byte, val rank: Byte, val rankInfo: RankInfo
) : Message
